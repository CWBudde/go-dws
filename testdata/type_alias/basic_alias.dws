// Basic type alias declarations
// Task 9.22: Test simple type aliases

type TUserID = Integer;
type TFileName = String;
type TPrice = Float;
type TFlag = Boolean;

// Test using aliases in variable declarations
var userID: TUserID := 12345;
var filename: TFileName := 'document.txt';
var price: TPrice := 99.99;
var flag: TFlag := True;

// Test assignment compatibility
var normalInt: Integer;
normalInt := userID;  // Should work - TUserID is compatible with Integer

// Print values to verify type aliases work
PrintLn('User ID: ' + IntToStr(normalInt));
PrintLn('Filename: ' + filename);
PrintLn('Price: 99.99');
PrintLn('Flag: true');
