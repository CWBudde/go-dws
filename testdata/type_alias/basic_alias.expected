User ID: 12345
Filename: document.txt
Price: 99.99
Flag: true
